wordle-web/scripts/game.ts

```diff
@@ -0,0 +1,38 @@
+import { Word } from "./word";
+
+export enum GameState {
+  Playing = 0,
+  Won = 1,
+  Loss = 2,
+}
+
+export class Game {
+  public maxAttempts: number = 6;
+  public guesses: Word[] = [];
+  public wordToGuess: string;
+  public state: GameState = GameState.Playing;
+
+  constructor(wordToGuess: string) {
+    this.wordToGuess = wordToGuess;
+  }
+
+  public guess(guess: string): void {
+    if (this.state !== GameState.Playing) {
+      return;
+    }
+
+    const word = new Word(guess);
+    const guessedWord = new Word(this.wordToGuess);
+    word.compare(guessedWord);
+    this.guesses.push(word);
+    this.updateGameState();
+  }
+
+  public updateGameState(): void {
+    if (this.guesses[this.guesses.length - 1].checkForWin() === true) {
```

Comment:
```suggestion
    if (this.guesses[this.guesses.length - 1].checkForWin()) {
```
Since we are using typescript and `checkForWin()` is typed boolean, we can use that alone for the condition without the extra comparison to `true`.

Bonus: Sometimes it can be helpful for readability to assign local variables values and use those in condition checks
```
let lastGuessCorrect = this.guesses[this.guesses.length - 1].checkForWin();
if (lastGuessCorrect){
```
While it is probably overkill here, it's a useful tool to keep in your back pocket when you start creating more complex functions.

Reply:
that's true, thank you!